Core/include/Acts/EventData/MultiComponentTrackParameters.hpp

```diff
@@ -67,11 +66,55 @@ class MultiComponentBoundTrackParameters {
   }
 
  public:
+  using ConstructionTuple = std::tuple<double, Acts::Vector4, Acts::Vector3,
+                                       double, CovarianceMatrix>;
+
+  /// We need this helper function in order to construct the base class properly
+  static MultiComponentBoundTrackParameters makeCurvilinear(
+      const std::vector<ConstructionTuple>& curvi,
+      ParticleHypothesis particleHypothesis) {
+    // TODO where to get a geometry context here
+    Acts::GeometryContext gctx{};
```

Comment:
This is only needed for the interface with the `planeSurface` that comes out of `CurvilinearSurface` right? I've had this issue a couple of times in the past, and the conclusion of various discussions was to still pipe it through the call-chain, even if you can prove locally that you can never be passing this to a surface that has a detector element attached. We've also discussed the possibility of just making it a runtime error if you pass an empty context and there is a detector element behind the surface.

In absence of this, I think this should be part of the function arguments.

Reply:
I agree - this was there before my changes. I can check how easy it is to change.